pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERRORS = (select.error, OSError, IOError)
```

Comment:
Thanks. Since I made the last push I realized that not all classes in _SELECT_ERROR_CHECKS needs to be in _SELECT_ERRORS if one is a subclass of another, only most generic ones need to be caught. I'll leave the optimization out for know and follow your suggestion.